sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/complexTypeExtractors.scala

```diff
@@ -148,7 +148,10 @@ trait ExtractValue extends Expression with QueryErrorsBase {
 case class GetStructField(child: Expression, ordinal: Int, name: Option[String] = None)
   extends UnaryExpression with ExtractValue {
 
-  lazy val childSchema = child.dataType.asInstanceOf[StructType]
```

Comment:
The assumption is that the child must be `StructType` so that we will construct `GetStructField`.